lib/network/level-net/index.js

```diff
@@ -0,0 +1,283 @@
+        dbClient[remoteCall] = function onCall(...args) {
+            const cb = args.pop();
+            args.push(this.path);
+            this.mdClient.callTimeout(remoteCall, args, cb,
+                                      this.callTimeoutMs);
+        };
+    }
+    for (const remoteCall of Object.keys(miscSyncCommands).concat(
+        Object.keys(miscAsyncCommands))) {
+        dbClient[remoteCall] = function onCall(...args) {
+            const cb = args.pop();
+            this.miscClient.callTimeout(remoteCall, args, cb,
+                                        this.callTimeoutMs);
+        };
+    }
+
+    /**
+     * @brief return a handle to a sublevel database
+     *
+     * @note this function has no side-effect on the db, it just
+     * returns a handle properly configured to access the sublevel db
+     * from the client.
+     *
+     * @param {String} subName name of sublevel
+     * @return {Object} a handle to the sublevel database that has the
+     * same API as its parent
+     */
+    dbClient.openSub = function openSub(subName) {
+        const subLevel = {};
+        for (const prop in this) {
+            if (!(prop in subLevel)) {
+                subLevel[prop] = this[prop];
+            }
+        }
+        // maintain path as a list of nested sublevels
+        subLevel.path = subLevel.path.slice();
+        subLevel.path.push(subName);
+        return subLevel;
+    };
+
+    return dbClient;
+};
+
+/**
+ * @brief create a server object that serves remote LevelDB requests
+ * through socket.io events
+ *
+ * @param {Object} db The root LevelDB database object to expose to
+ * remote clients
+ * @param {Object} options currently supported options are:
+ * - streamMaxPendingAck {Number} max number of in-flight output
+ *   stream packets sent to the client without an ack received yet
+ * - streamAckTimeoutMs {Number} timeout for receiving an ack after an
+ *   output stream packet is sent to the client
+ * @return {Object} a server object, not yet listening on a TCP port
+ * (call listen() next for this)
+ */
+module.exports.createServer = function LevelServer(db, options = {}) {
+    const httpServer = http.createServer();
+    const ioServer = io(httpServer);
+
+    ioServer.initMetadataService = function initMetadataService(baseNs = '') {
+        assert(baseNs === '' || baseNs.startsWith('/'));
+
+        const mdSock = this.of(`${baseNs}/metadata`);
+        mdSock.on('connection', conn => {
+            function openSub(path) {
+                let subDb = db;
+                path.forEach(pathItem => {
+                    subDb = subDb.sublevel(pathItem);
+                });
+                return subDb;
+            }
+
+            const mdStreams = sioStream(conn,
+                                        options.streamMaxPendingAck,
+                                        options.streamAckTimeoutMs);
+            conn.on('error', () => {});
```

Comment:
We could log something here though in the current code index does not have a werelogs logging context, and it shall be logged in an upper level. We have to have a listener for errors though otherwise it's going to throw.